app/src/main/AndroidManifest.xml

```diff
@@ -201,7 +201,8 @@
           android:value="main.MainActivity"/>
     </activity>
     <activity android:name=".zim_manager.local_file_transfer.LocalFileTransferActivity"
-        android:label="Send to nearby device">
+        android:label="Send to nearby device"
+        android:screenOrientation="portrait"> <!--Protect AsyncTask from orientation changes-->
```

Comment:
I would delete the comment.
Also "protect asyntask" is a bit naïve, it is very easy for an activity to be killed and recreated, just the most common way is by rotation but it happens on any "configuration change" like language. You can also simulate it by turning on "don't keep activities" in Developer Settings and pressing the home button while asynctask is in progress.
The more modern approach is to have your asynchronous processing live in your viewModel which is an object that persists between configuration changes. The viewmodel holds data, the view observes changes in that data and sends actions to the viewModel which may resolve these actions into updates to its data.

Reply:
I meant to answer the query "why is the orientation fixed here"
Should I update it to `<!--To protect AsyncTask from orientation changes. Still susceptible to configuration changes.-->` or remove it altogether?

Reply:
Remove it altogether and make sure you cancel your Asynctasks in onDestroy to prevent activity leaks

Reply:
Done - Added a `cancelAsyncTasks()` method [here](https://github.com/kiwix/kiwix-android/pull/1276/commits/73268cfb5bae3e83eae813de4925ad2a7bde015a) and shifted it to onDestroy() [here](https://github.com/kiwix/kiwix-android/pull/1276/commits/d4ffe18ad8bfe865ca93aba890121e5868b10a81)